tests/tests/compiler/type_inference/basic_inference.leo

```diff
@@ -0,0 +1,29 @@
+program foo.aleo {
+    struct S {
+        x: u32,
+        y: u32,
+    }
+
+    record R {
+        owner: address,
+        v: u64
+    }
+    
+    transition main(x: u32, y: u32, addr: address, arr: [u32; 4]) 
+        -> (S, S, u64, u32, u32, bool, u32, u32) {
+        let s1 = S { x, y };
+        let s2: S = s1;
```

Comment:
Just curious, does inference work if we remove this type annotation?

Reply:
Yes. The next line is `let s3 = s1;` which is the same but without the annotation and it works fine!